Sources/SwiftRefactor/CallToTrailingClosures.swift

```diff
@@ -0,0 +1,115 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftSyntax
+
+/// Convert a call with inline closures to one that uses trailing closure
+/// syntax. Returns `nil` if there's already trailing closures or there are no
+/// closures within the call. Pass `conversionStart` to specify an index to
+/// start the conversion from, ie. to skip converting closures before
+/// `conversionStart`
+///
+/// ## Before
+/// ```
+/// someCall(closure1: (String) -> Int, closure2: (String) -> Int)
+/// ```
+///
+/// ## After
+/// ```
+/// someCall { arg in
+///   <#code#>
+/// } closure2: { arg in
+///   <#code#>
+/// }
+/// ```
+public struct CallToTrailingClosures: RefactoringProvider {
+  public struct Options {
+    public let startAt: Int
+
+    public init(startAt: Int = 0) {
+      self.startAt = startAt
+    }
+  }
+  public typealias Context = Options
+
+  // TODO: Rather than returning nil, we should consider throwing errors with
+  // appropriate messages instead.
+  public static func refactor(syntax call: FunctionCallExprSyntax, in context: Context = Context()) -> FunctionCallExprSyntax? {
+    return call.convertToTrailingClosures(from: context.startAt)
+  }
+}
+
+extension FunctionCallExprSyntax {
+  fileprivate func convertToTrailingClosures(from startAt: Int) -> FunctionCallExprSyntax? {
+    guard trailingClosure == nil, additionalTrailingClosures == nil, leftParen != nil, rightParen != nil else {
+      // Already have trailing closures
+      return nil
+    }
+
+    var closures = [(label: TokenSyntax, closure: ClosureExprSyntax)]()
+    for arg in argumentList.dropFirst(startAt) {
+      guard let closure = arg.expression.as(ClosureExprSyntax.self) else {
+        closures.removeAll()
+        continue
+      }
+
+      // Trailing comma won't exist any more, move its trivia to the end of
+      // the closure instead
+      closures.append((arg.label ?? .wildcardToken(), closure.with(\.trailingTrivia, (closure.trailingTrivia ?? []) + (arg.trailingComma?.trailingTrivia ?? []))))
```

Comment:
Technically, you would also need to move the trailing comma’s leading trivia, right? I think you could end up with a comment as leading trivia if you have the comma on a newline like the following where the comma is on a newline (haven’t verified though). Maybe just retain the comments from the trivia?
```swift
foo(1
/*some comment*/,2)
```

Reply:
Yes, There's probably a few places like this - I was waiting for https://github.com/apple/swift-syntax/pull/1478 (which was waiting on the other). I'll fix that up and then use it here.

Reply:
> Maybe just retain the comments from the trivia?

I'd rather keep whitespace I think, ie.
```
foo(1 /* some comments */
/*more comments*/, { _ in })
```
->
```
foo(1/* some comments */
/*more comments*/) { _ in }
```

Rather than:
```
foo(1/* some comments *//*more comments*/) { _ in }
```

Reply:
I was just thinking to strip whitespace to avoid the need to do more complicated trivia merging. But it’s good with me if you’re planning to update this once #1478 is merged.